okhttp/src/main/java/okhttp3/internal/http2/Hpack.java

```diff
@@ -511,6 +520,36 @@ void writeInt(int value, int prefixMask, int bits) {
       out.writeByte(value);
     }
 
+    int getHeaderIndex(Header header) {
+      int dynamicHeaderIndex = Util.indexOf(dynamicTable, header);
+      if (dynamicHeaderIndex != -1) return normalizeDynamicTableIndex(dynamicHeaderIndex);
+
+      // Only search a subset of the static header table. Most entries have an empty value, so it's
+      // unnecessary to waste cycles looking at them.
+      for (int i = 1; i < 7; i++) {
+        if (equal(STATIC_HEADER_TABLE[i], header)) return i + 1;
+      }
+
+      return -1;
+    }
+
+    /** Converts a raw array index in the dynamic table array into the HPACK index address space. */
+    int normalizeDynamicTableIndex(int arrayIndex) {
+      return arrayIndex - nextHeaderIndex + STATIC_HEADER_TABLE.length;
+    }
+
+    int getHeaderNameIndex(ByteString name) {
+      if (NAME_TO_FIRST_INDEX.containsKey(name)) return NAME_TO_FIRST_INDEX.get(name) + 1;
```

Comment:
Call `get()` and check for `null` to avoid having to do double hashing and lookup.